class-tgm-plugin-activation.php

```diff
@@ -1286,25 +1337,25 @@ protected function _gather_plugin_data() {
 			$i                 = 0;
 			$installed_plugins = get_plugins();
 
-			foreach ( $this->tgmpa->plugins as $plugin ) {
-				if ( is_plugin_active( $plugin['file_path'] ) || ( isset( $plugin['is_callable'] ) && is_callable( $plugin['is_callable'] ) ) ) {
+			foreach ( $this->tgmpa->plugins as $slug => $plugin ) {
+				if ( is_plugin_active( $plugin['file_path'] ) || ( ! empty( $plugin['is_callable'] ) && is_callable( $plugin['is_callable'] ) ) ) {
 					continue; // No need to display plugins if they are installed and activated.
 				}
 
 				$table_data[ $i ]['sanitized_plugin'] = $plugin['name'];
-				$table_data[ $i ]['slug']             = $this->_get_plugin_data_from_name( $plugin['name'] );
-
-				$external_url = $this->_get_plugin_data_from_name( $plugin['name'], 'external_url' );
-				$source       = $this->_get_plugin_data_from_name( $plugin['name'], 'source' );
+				$table_data[ $i ]['slug']             = $slug;
 
-				if ( $external_url && preg_match( '|^http(s)?://|', $external_url ) ) {
-					$table_data[ $i ]['plugin'] = '<strong><a href="' . esc_url( $external_url ) . '" target="_blank">' . esc_html( $plugin['name'] ) . '</a></strong>';
-				}
-				elseif ( ! $source || preg_match( '|^http://wordpress.org/extend/plugins/|', $source ) ) {
+				if ( ! empty( $plugin['external_url'] ) && preg_match( TGM_Plugin_Activation::EXT_REPO_REGEX, $plugin['external_url'] ) ) {
```

Comment:
I'm abstracting this (whole block of code) in a different way in one of the new branches